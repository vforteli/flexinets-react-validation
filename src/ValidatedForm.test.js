import React from 'react';
import ValidatedForm from './ValidatedForm';
import TextInputValidated from './TextInputValidated';

test('ValidatedForm renders without errors', () => {
    const wrapper = mount(
        <ValidatedForm>
            <TextInputValidated type="text" name="test" label="test" required placeholder="test" />
        </ValidatedForm>
    );

    expect(wrapper).toMatchSnapshot();
});

test('ValidatedForm renders with validation error', () => {
    const wrapper = mount(
        <ValidatedForm>
            <TextInputValidated type="text" name="test" label="test" required placeholder="test" />
            <button type="submit" id="submitbutton">submit</button>
        </ValidatedForm>
    );

    // todo fix this test...
    wrapper.find('#submitbutton').simulate("click");
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
});


// todo add more tests