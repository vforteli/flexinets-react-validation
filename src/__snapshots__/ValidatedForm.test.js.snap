// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidatedForm renders with validation error 1`] = `
<ValidatedForm>
  <form
    noValidate={true}
    onSubmit={[Function]}
  >
    <_class
      label="test"
      name="test"
      placeholder="test"
      required={true}
      type="text"
    >
      <TextInputValidated
        isFormTouched={false}
        label="test"
        name="test"
        placeholder="test"
        required={true}
        type="text"
      >
        <div
          className="form-group required"
        >
          <label
            htmlFor="test"
          >
            test
             
          </label>
          <input
            className="form-control"
            label="test"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="test"
            required={true}
            type="text"
          />
          <div
            className="invalid-feedback"
          />
        </div>
      </TextInputValidated>
    </_class>
    <button
      id="submitbutton"
      type="submit"
    >
      submit
    </button>
  </form>
</ValidatedForm>
`;

exports[`ValidatedForm renders without errors 1`] = `
<ValidatedForm>
  <form
    noValidate={true}
    onSubmit={[Function]}
  >
    <_class
      label="test"
      name="test"
      placeholder="test"
      required={true}
      type="text"
    >
      <TextInputValidated
        isFormTouched={false}
        label="test"
        name="test"
        placeholder="test"
        required={true}
        type="text"
      >
        <div
          className="form-group required"
        >
          <label
            htmlFor="test"
          >
            test
             
          </label>
          <input
            className="form-control"
            label="test"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="test"
            required={true}
            type="text"
          />
          <div
            className="invalid-feedback"
          />
        </div>
      </TextInputValidated>
    </_class>
  </form>
</ValidatedForm>
`;
